src/ray/core_worker/transport/out_of_order_actor_scheduling_queue.h

```diff
@@ -92,11 +96,19 @@ class OutOfOrderActorSchedulingQueue : public SchedulingQueue {
   bool is_asyncio_ = false;
   /// Mutext to protect attributes used for thread safe APIs.
   absl::Mutex mu_;
+  /// This stores all the tasks that have previous attempts that are pending.
+  /// They are queued and will be executed after the previous attempt finishes.
+  /// This can happen if transient network error happens after an actor
+  /// task is submitted and recieved by the actor and the caller retries
+  /// the same task.
+  absl::flat_hash_map<TaskID, InboundRequest> queued_actor_tasks_ ABSL_GUARDED_BY(mu_);
```

Comment:
We may want to change pending_task_id_to_is_canceled value to a state enum.
```cc
enum TaskState {
// Waiting for deps, can be cancelled.
// Invariant: queued_actor_tasks_[task_id] does not exist.
WAITING_FOR_DEPS = 0;
// Waiting for deps, but cancelled by user.
CANCELLED_BY_USER;
// Waiting for deps, but cancelled by newer attempts.
CANCELLED_BY_NEW_ATTEMPTS;
// Running, can't be cancelled.
RUNNING;
};
```

On cancel:
- case WAITING_FOR_DEPS, CANCELLED_BY_NEW_ATTEMPTS: change to CANCELLED_BY_USER, cancel queued_actor_tasks_[task_id]
- else: cancel queued_actor_tasks_[task_id]

On new attempt Add:
- case WAITING_FOR_DEPS: change to CANCELLED_BY_NEW_ATTEMPTS, add new attempt to queued_actor_tasks_
- case CANCELLED_BY_USER: also cancel new attempt.
- else: add new attempt to queued_actor_tasks_

On dep resolved:
- case WAITING_FOR_DEPS: run
- case CANCELLED_*: Cancel(reason)
- case RUNNING: check fail

On run finished:
- case RUNNING: erase the state, wait-dep for queued_actor_tasks_[task_id] if any
- else: check fail

Reply:
`pending_task_id_to_is_canceled` cancel here only means user triggered cancellation.